WordPress/src/main/java/org/wordpress/android/ui/stats/service/StatsService.java

```diff
@@ -136,108 +135,124 @@ private void stopRefresh() {
                 }
             }
             mStatsNetworkRequests.clear();
-            mNumberOfFinishedNetworkCalls = 0;
-            mNumberOfNetworkCalls = 0;
         }
     }
 
-    private void startTasks(final String blogId, final StatsTimeframe timeframe, final String date,
-                            final boolean updateGraphStats, final boolean updateAlltimeStats, final int startId) {
-        this.mServiceBlogId = blogId;
-        this.mServiceRequestedTimeframe = timeframe;
-        this.mServiceRequestedDate = date;
-        this.mServiceStartId = startId;
+    private void startTasks(final String blogId, final StatsTimeframe timeframe, final String date, final StatsEndpointsEnum sectionToUpdate,
+                            final int maxResultsRequested, final int pageRequested) {
 
-        new Thread() {
-            @Override
-            public void run() {
-                final RestClientUtils restClientUtils = WordPress.getRestClientUtilsV1_1();
+        final RestClientUtils restClientUtils = WordPress.getRestClientUtilsV1_1();
 
-                String period = timeframe.getLabelForRestCall();
+        String period = timeframe.getLabelForRestCall();
+        /*AppLog.i(T.STATS, "A new Stats network request is required for blogID: " + blogId + " - period: " + period
+                + " - date: " + date + " - StatsType: " + sectionToUpdate.name());
+*/
+        EventBus.getDefault().post(new StatsEvents.UpdateStatusChanged(true));
 
-                AppLog.i(T.STATS, "Update started for blogID - " + blogId + " with the following period: " + period
-                        + " on the following date: " + mServiceRequestedDate);
+        RestListener vListener = new RestListener(sectionToUpdate, blogId, timeframe, date);
+        String path = String.format("/sites/%s/stats/", blogId);
+        synchronized (mStatsNetworkRequests) {
+            switch (sectionToUpdate) {
+                case VISITS:
+                    path = String.format(path + "visits?unit=%s&quantity=10&date=%s", period, date);
```

Comment:
I've cleaned the code a bit. Will run the app during the weekend and open a PR on monday if everything goes fine on my tests.